qiskit/ignis/verification/randomized_benchmarking/fitters.py

```diff
@@ -750,3 +763,189 @@ def plot_rb_data(self, pattern_index=0, ax=None,
+        and another RBFitter object:
+        calculates their sum and does the fit
+    """
+
+    def __init__(self, purity_result, npurity, cliff_lengths,
+                 rb_pattern=None):
+        """
+        Args:
+            purity_result: list of lists of results of the
+            3^n purity RB sequences per seed (qiskit.Result).
+            npurity: equals 3^n (where n is the dimension)
+            cliff_lengths: the Clifford lengths, 2D list i x j where i is the
+                number of patterns, j is the number of cliffords lengths
+            rb_pattern: the pattern for the rb sequences.
+        """
+
+        self._cliff_lengths = cliff_lengths
+        self._rb_pattern = rb_pattern
+        self._npurity = npurity
+        self._fit = [{} for e in rb_pattern]
+
+        self._rbfit_purity = [[] for _ in range(npurity+1)]
+        # 3^n RBFitter objects
+        for d in range(self._npurity):
+            self._rbfit_purity[d] = RBFitter(
+                purity_result[d], cliff_lengths, rb_pattern)
+        # The last RBFitter is the sum of 3^n correlators
+        self._rbfit_purity[self._npurity] = RBFitter(
+            None, cliff_lengths, rb_pattern)
+
+        self.add_data(purity_result)
+
+    @property
+    def rbfit_pur(self):
+        """Return the purity RB fitter."""
+        return self._rbfit_purity
+
+    @property
+    def cliff_lengths(self):
+        """Return clifford lengths."""
+        return self._cliff_lengths
+
+    @property
+    def fit(self):
+        """Return the purity fit parameters."""
+        return self.rbfit_pur[self._npurity].fit
+
+    @property
+    def fit_pur(self):
+        """Return the fit as a 3^n element list."""
+        return [self.rbfit_pur[d].fit for
+                d in range(self._npurity)]
+
+    @property
+    def rb_fit_fun(self):
+        """Return the function rb_fit_fun."""
+        return self.rbfit_pur[self._npurity] \
+            .rb_fit_fun
+
+    @property
+    def seeds(self):
+        """Return the number of loaded seeds as a
+        3^n element list."""
+        return [self.rbfit_pur[d].seeds for
+                d in range(self._npurity)]
+
+    @property
+    def results(self):
+        """Return all the results as a 3^n element list."""
+        return [self.rbfit_pur[d].results for
+                d in range(self._npurity)]
+
+    @property
+    def raw_data_pur(self):
+        """Return raw_data as a
+        3^n element list."""
+        return [self.rbfit_pur[d].raw_data for
+                d in range(self._npurity)]
+
+    @property
+    def raw_data(self):
+        """Return raw_data of the sum of 3^n
+        correlators"""
+        return self.rbfit_pur[self._npurity].raw_data
+
+    @property
+    def ydata(self):
+        """Return ydata (means and std devs)
+        of the sum of 3^n correlators"""
+        return self.rbfit_pur[self._npurity].ydata
+
+    def add_data(self, new_purity_result, rerun_fit=True):
```

Comment:
This assumes too much structure in the result. It might be that you want to bundle some of the purity operations into a single job (in which case there will be only one result and it won't be divided). You need this structure because you trying to cleaning send to the 3^n RBFitters, so I'd scrap that. It's not helping enough.

Reply:
We need all 3^n results (from each seed and length) to calculate the fitter, so it's easier to assume that we have 3^n array.

Reply:
(even if the job is executed differently, the results can be arranged in an array afterwards)

Reply:
OK - I changed this